backends/bmv2/common/parser.cpp

```diff
@@ -48,6 +48,61 @@ cstring ParserConverter::jsonAssignment(const IR::Type* type, bool inParser) {
 Util::IJson* ParserConverter::convertParserStatement(const IR::StatOrDecl* stat) {
     auto result = new Util::JsonObject();
     auto params = mkArrayField(result, "parameters");
+    auto isR = false;
+    IR::MethodCallExpression *mce2 = nullptr;
+    if (stat->is<IR::AssignmentStatement>()) {
+        auto assign = stat->to<IR::AssignmentStatement>();
+        if (assign->right->is<IR::MethodCallExpression>()) {
+            auto mce = assign->right->to<IR::MethodCallExpression>();
+            auto minst = P4::MethodInstance::resolve(mce, ctxt->refMap, ctxt->typeMap);
+            if (minst->is<P4::ExternMethod>()) {
+                auto extmeth = minst->to<P4::ExternMethod>();
+                if ((extmeth->method->name.name == "get"
+                    || extmeth->method->name.name == "get_state")
+                    && extmeth->originalExternType->name == "InternetChecksum") {
+                    const IR::Expression *l;
+                    l = assign->left;
+                    isR=true;
+                    auto dest = new IR::Argument(l);
+                    auto args = new IR::Vector<IR::Argument>();
+                    args->push_back(dest);  // dest
+                    mce2 = new IR::MethodCallExpression(mce->method, mce->typeArguments);
+                    mce2->arguments = args;
+                    stat = new IR::MethodCallStatement(mce);
+                }
+            }
+        } else if (assign->right->is<IR::Equ>()
```

Comment:
this is awfully specific. 
what happens to some equivalent rewrites of this expression, e.g., assigning the result to a temporary and then comparing?

Reply:
This is used when **get** method is in **verify** functin. For example **verify(ipv4.hdrChecksum == ck.get(), flag)**. Compiler will always transform this in:

**bool tmp;
tmp = ipv4.hdrChecksum == ck.get(); 
verify(tmp, flag)**

Equ expression is for the second line. To process **tmp** and **ipv4.hdrChecksum** for bm **PSA_InternetChecksum::get_verify()** method.

Reply:
For example, what if the compiler generates:
```
a = ck.get();
verify(a == ipv4.checksum, "")
```
This is a perfectly legal transformation.

Reply:
There are two possible cases.

1. If the temporary variable `a` is assigned in the same block as `verify` function. 
 ```
a=ck.get();
 verify(hdr.ipv4.hdrChecksum == ck.get(), error.BadIPv4HeaderChecksum);
```

Then the compiler will transform it before it gets in this part of code:
![Screenshot from 2021-08-13 09-30-01](https://user-images.githubusercontent.com/77269536/129320858-fc92f57a-6d20-4063-b880-a4c716a48142.png)
This case has been resolved with this `Equ` `if clause` in `p4c` and with `get_verify(Field &dst, Field &equOp) const;` method in bm project.
First it asks if there is an `Equ` on the right side of `=`. If it is than it asks if there is an extern method. If there is it will precess it only if there is `InternetChecksum::get`.

2. If the temporary variable `a` is assigned in the other block in parser.
```
state x {
...
a = ck.get();
...
}
state y {
...
verify(hdr.ipv4.hdrChecksum == a, error.BadIPv4HeaderChecksum);
...
}
```
Then the compiler will transform it:
![Screenshot from 2021-08-13 09-38-05](https://user-images.githubusercontent.com/77269536/129321753-0fa7b0fd-bc53-465a-99b6-567ee3e8909e.png)
Here there is no `get` method in `Equ` and "this is no more extern method `get` problem". 
This will be processed in classic `Assignment statement`:
https://github.com/p4lang/p4c/pull/2872/files#diff-d0c07a9f6f6d88c535125cb0a83255a81340b8f55f87f645b0bfcbf58109ea0eR106-R126
In JSON this case looks like this (`tmp_10 = hdr.ipv4.hdrChecksum == a_0;`):
![Screenshot from 2021-08-13 09-50-22](https://user-images.githubusercontent.com/77269536/129323241-5871fa1c-935e-4f96-a37e-d7eac1d30226.png)
So in bm it is not `get` method problem.

Reply:
This code is not incorrect, it is just overly specific. I hope it won't break when other changes are made to passes that precede it.